From 4d37ba2231b6eea5c44e07c0380b2963ffa0de47 Mon Sep 17 00:00:00 2001
From: Shivananda Hebbar <x0hebbar@ti.com>
Date: Fri, 19 Mar 2010 15:12:03 -0600
Subject: [PATCH 071/135] DSPBRIDGE: Change channel mode type to s8

chnl_mode or dw_mode can have value 0 or 1.
Hence changing the type to s8.

Signed-off-by: Shivananda Hebbar <x0hebbar@ti.com>
---
 arch/arm/plat-omap/include/dspbridge/_chnl_sm.h |    2 +-
 arch/arm/plat-omap/include/dspbridge/chnldefs.h |    4 ++--
 arch/arm/plat-omap/include/dspbridge/chnlpriv.h |    2 +-
 arch/arm/plat-omap/include/dspbridge/wmd.h      |    2 +-
 arch/arm/plat-omap/include/dspbridge/wmdchnl.h  |    2 +-
 drivers/dsp/bridge/rmgr/node.c                  |    2 +-
 drivers/dsp/bridge/rmgr/strm.c                  |    2 +-
 drivers/dsp/bridge/wmd/chnl_sm.c                |    8 ++++----
 8 files changed, 12 insertions(+), 12 deletions(-)

diff --git a/arch/arm/plat-omap/include/dspbridge/_chnl_sm.h b/arch/arm/plat-omap/include/dspbridge/_chnl_sm.h
index 5f89fc6..bc09ee0 100644
--- a/arch/arm/plat-omap/include/dspbridge/_chnl_sm.h
+++ b/arch/arm/plat-omap/include/dspbridge/_chnl_sm.h
@@ -143,7 +143,7 @@ struct chnl_object {
 	struct chnl_mgr *chnl_mgr_obj;
 	u32 chnl_id;		/* Channel id */
 	u8 dw_state;		/* Current channel state */
-	u32 chnl_mode;		/* Chnl mode and attributes */
+	s8 chnl_mode;		/* Chnl mode and attributes */
 	/* Chnl I/O completion event (user mode) */
 	bhandle user_event;
 	/* Abstract syncronization object */
diff --git a/arch/arm/plat-omap/include/dspbridge/chnldefs.h b/arch/arm/plat-omap/include/dspbridge/chnldefs.h
index a12870e..66fb660 100644
--- a/arch/arm/plat-omap/include/dspbridge/chnldefs.h
+++ b/arch/arm/plat-omap/include/dspbridge/chnldefs.h
@@ -26,8 +26,8 @@
 #define CHNL_INITIOREQS      4	/* Default # of I/O requests. */
 
 /* Channel modes */
-#define CHNL_MODETODSP       0x0000	/* Data streaming to the DSP. */
-#define CHNL_MODEFROMDSP     0x0001	/* Data streaming from the DSP. */
+#define CHNL_MODETODSP		0	/* Data streaming to the DSP. */
+#define CHNL_MODEFROMDSP	1	/* Data streaming from the DSP. */
 
 /* GetIOCompletion flags */
 #define CHNL_IOCINFINITE     0xffffffff	/* Wait forever for IO completion. */
diff --git a/arch/arm/plat-omap/include/dspbridge/chnlpriv.h b/arch/arm/plat-omap/include/dspbridge/chnlpriv.h
index f58f7d9..750b791 100644
--- a/arch/arm/plat-omap/include/dspbridge/chnlpriv.h
+++ b/arch/arm/plat-omap/include/dspbridge/chnlpriv.h
@@ -77,7 +77,7 @@ struct chnl_info {
 	bhandle event_obj;	/* Channel I/O completion event. */
 	/*Abstraction of I/O completion event. */
 	struct sync_object *sync_event;
-	u32 dw_mode;		/* Channel mode. */
+	s8 dw_mode;		/* Channel mode. */
 	u8 dw_state;		/* Current channel state. */
 	u32 bytes_tx;		/* Total bytes transferred. */
 	u32 cio_cs;		/* Number of IOCs in queue. */
diff --git a/arch/arm/plat-omap/include/dspbridge/wmd.h b/arch/arm/plat-omap/include/dspbridge/wmd.h
index 149837c..38566f2 100644
--- a/arch/arm/plat-omap/include/dspbridge/wmd.h
+++ b/arch/arm/plat-omap/include/dspbridge/wmd.h
@@ -409,7 +409,7 @@ typedef void (*fxn_deh_notify) (struct deh_mgr *hdeh_mgr,
 typedef dsp_status(*fxn_chnl_open) (OUT struct chnl_object
 				    **phChnl,
 				    struct chnl_mgr *hchnl_mgr,
-				    short int chnl_mode,
+				    s8 chnl_mode,
 				    u32 uChnlId,
 				    CONST IN OPTIONAL struct
 				    chnl_attr * pattrs);
diff --git a/arch/arm/plat-omap/include/dspbridge/wmdchnl.h b/arch/arm/plat-omap/include/dspbridge/wmdchnl.h
index 0d874ab..57ca2af 100644
--- a/arch/arm/plat-omap/include/dspbridge/wmdchnl.h
+++ b/arch/arm/plat-omap/include/dspbridge/wmdchnl.h
@@ -33,7 +33,7 @@ extern dsp_status bridge_chnl_destroy(struct chnl_mgr *hchnl_mgr);
 
 extern dsp_status bridge_chnl_open(OUT struct chnl_object **phChnl,
 				   struct chnl_mgr *hchnl_mgr,
-				   short int chnl_mode,
+				   s8 chnl_mode,
 				   u32 uChnlId,
 				   CONST IN OPTIONAL struct chnl_attr
 				   *pattrs);
diff --git a/drivers/dsp/bridge/rmgr/node.c b/drivers/dsp/bridge/rmgr/node.c
index 78db5a3..7250733 100644
--- a/drivers/dsp/bridge/rmgr/node.c
+++ b/drivers/dsp/bridge/rmgr/node.c
@@ -849,7 +849,7 @@ dsp_status node_connect(struct node_object *hNode1, u32 uStream1,
 	struct stream_chnl *pstream;
 	gb_bit_num pipe_id = GB_NOBITS;
 	gb_bit_num chnl_id = GB_NOBITS;
-	short int chnl_mode;
+	s8 chnl_mode;
 	u32 dw_length;
 	dsp_status status = DSP_SOK;
 	DBC_REQUIRE(refs > 0);
diff --git a/drivers/dsp/bridge/rmgr/strm.c b/drivers/dsp/bridge/rmgr/strm.c
index aa27e08..aabc1d5 100644
--- a/drivers/dsp/bridge/rmgr/strm.c
+++ b/drivers/dsp/bridge/rmgr/strm.c
@@ -480,7 +480,7 @@ dsp_status strm_open(struct node_object *hnode, u32 dir, u32 index,
 	struct bridge_drv_interface *intf_fxns;
 	u32 ul_chnl_id;
 	struct strm_object *strm_obj = NULL;
-	short int chnl_mode;
+	s8 chnl_mode;
 	struct chnl_attr chnl_attr_obj;
 	dsp_status status = DSP_SOK;
 	struct cmm_object *hcmm_mgr = NULL;	/* Shared memory manager hndl */
diff --git a/drivers/dsp/bridge/wmd/chnl_sm.c b/drivers/dsp/bridge/wmd/chnl_sm.c
index 595981d..919a75f 100644
--- a/drivers/dsp/bridge/wmd/chnl_sm.c
+++ b/drivers/dsp/bridge/wmd/chnl_sm.c
@@ -257,7 +257,7 @@ dsp_status bridge_chnl_cancel_io(struct chnl_object *chnl_obj)
 	dsp_status status = DSP_SOK;
 	struct chnl_object *pchnl = (struct chnl_object *)chnl_obj;
 	u32 chnl_id = -1;
-	short int chnl_mode;
+	s8 chnl_mode;
 	struct chnl_irp *chnl_packet_obj;
 	struct chnl_mgr *chnl_mgr_obj = NULL;
 
@@ -490,7 +490,7 @@ dsp_status bridge_chnl_flush_io(struct chnl_object *chnl_obj, u32 dwTimeOut)
 {
 	dsp_status status = DSP_SOK;
 	struct chnl_object *pchnl = (struct chnl_object *)chnl_obj;
-	short int chnl_mode = -1;
+	s8 chnl_mode = -1;
 	struct chnl_mgr *chnl_mgr_obj;
 	struct chnl_ioc chnl_ioc_obj;
 	/* Check args: */
@@ -756,7 +756,7 @@ dsp_status bridge_chnl_get_mgr_info(struct chnl_mgr *hchnl_mgr, u32 uChnlID,
 dsp_status bridge_chnl_idle(struct chnl_object *chnl_obj, u32 dwTimeOut,
 			    bool fFlush)
 {
-	short int chnl_mode;
+	s8 chnl_mode;
 	struct chnl_mgr *chnl_mgr_obj;
 	dsp_status status = DSP_SOK;
 
@@ -784,7 +784,7 @@ dsp_status bridge_chnl_idle(struct chnl_object *chnl_obj, u32 dwTimeOut,
  *      Open a new half-duplex channel to the DSP board.
  */
 dsp_status bridge_chnl_open(OUT struct chnl_object **phChnl,
-			    struct chnl_mgr *hchnl_mgr, short int chnl_mode,
+			    struct chnl_mgr *hchnl_mgr, s8 chnl_mode,
 			    u32 uChnlId, CONST IN struct chnl_attr *pattrs)
 {
 	dsp_status status = DSP_SOK;
-- 
1.7.0.4

